src/DDTrace/Integrations/Symfony/SymfonyIntegration.php

```diff
@@ -366,10 +388,62 @@ function () use ($integration) {
             }
         );
 
+        $handle_route = function($request, $rootSpan) use ($integration) {
+            $route_name = $request->get('_route');
+            if ($route_name === null) {
+                return;
+            }
+            if (dd_trace_env_config("DD_HTTP_SERVER_ROUTE_BASED_NAMING")) {
+                $rootSpan->resource = $route_name;
+            }
+            $rootSpan->meta['symfony.route.name'] = $route_name;
+
+            // the rest is for determining http.route
```

Comment:
I'd rather extract then all the code below this comment to its own function. Comments sooner or later lie. Just my opinion, not blocking the pr or anything

Reply:
I've refactored it